transaction/core/src/amount.rs

```diff
@@ -30,7 +33,7 @@ pub enum AmountError {
 pub struct Amount {
     /// A Pedersen commitment `v*G + b*H` to a quantity `v` of MobileCoin, with blinding `b`,
     #[prost(message, required, tag = "1")]
-    pub commitment: Commitment,
+    pub commitment: CompressedCommitment,
```

Comment:
Previously, "Commitment" was a type alias for CurvePoint, which we used for any RistrettoPoint that we wanted to serialize.

This now uses the newtype CompressedCommitment which wraps CompressedRistretto. The dedicated newtype gives better type checking, and I think that storing this as a compressed point avoids ambiguity and is more performant (similar to how KeyImages are represented as compressed points).